docs/how-to-integrate-a-frontend/how-to-integrate-dotwallet.md

```diff
@@ -0,0 +1,196 @@
+      process.env.REDIRECT_URI || ''
+    )}&response_type=code&scope=${encodeURIComponent(
+      "user.info"
+    )}&state=${crypto.randomUUID()}`;
+  
+    return (
+      <main className="flex min-h-screen flex-col items-center justify-between p-24">
+        <div className="m-4 p-4 bg-blue-200 font-bold rounded-lg">
+          <a href={loginUrl}>DotWallet Login</a>
+        </div>
+      </main>
+    );
+}
+```
+
+<center>src/app/page.tsx</center>
+
+
+If the user clicks the **DotWallet Login** button, the page will be redirected to the authorization server page.
+
+![](../../static/img/DotWallet-auth.png)
+
+
+2. After the user clicks **Agree to authorize** to log in, the authorization server redirect the user to the redirection URI. The following code receives the `code` through the callback uri. Then exchange the `code` for `access_token` and save it.
+
+```ts
+import { redirect, notFound } from 'next/navigation';
+
+import token from "../token"
+
+export async function GET(request: Request) {
+  const { searchParams } = new URL(request.url);
+  const code = searchParams.get('code');
+
+  if (code) {
+    // exchange the code for access_token
+    const res = await fetch(`https://api.ddpurse.com/v1/oauth2/get_access_token`, {
+      body: JSON.stringify({
+        code,
+        redirect_uri: process.env.REDIRECT_URI,
+        grant_type: "authorization_code",
+        client_secret: process.env.CLIENT_SECRET,
+        client_id: process.env.CLIENT_ID,
+      }),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      method: 'POST'
+    });
+    const { code: apiCode, data, msg } = await res.json();
+
+    if (apiCode === 0) {
+      const { access_token } = data;
+      // save access_token
+      token.access_token = access_token;
+      // redirect to balance page.
+      redirect('/balance');
+    }
+
+  }
+
+  notFound();
+}
+```
+
+<center>src/app/callback/route.ts</center>
+
+
+### `DotWalletSigner`
+
+sCrypt SDK provides `DotWalletSigner` for quick integration with DotWallet.
+
+After redirect to the `/balance` page, we can create a `DotWalletSigner` with the OAuth access token. 
+
+
+
+```ts
+import { DotwalletSigner, DefaultProvider } from "scrypt-ts";
+import token from "../token";
+
+async function getData() {
+  const provider = new DefaultProvider();
+  const signer = new DotwalletSigner(token.access_token, provider);
+
+  const balance = await signer.getBalance();
+
+  return { balance: balance.confirmed + balance.unconfirmed };
+}
+
+export default async function Balance() {
+  const data = await getData();
+
+  return (
+    <main className="flex min-h-screen flex-col items-center justify-between p-24">
+      <div className="m-4 p-4 bg-blue-200 font-bold rounded-lg">
+        <label>balance</label> {data.balance}
+      </div>
+    </main>
+  );
+}
+
+```
+
+<center>src/app/balance/page.tsx</center>
```

Comment:
Again, explain how `redirect('/balance')` redirects to `page.tsx`.

Reply:
see above